adhocracy4/comments_async/serializers.py

```diff
@@ -58,6 +52,41 @@ def to_representation(self, instance):
         ret['comment_categories'] = categories
         if instance.is_blocked:
             ret['comment'] = ''
+
+        request = self.context.get('request')
+        ret['is_users_own_comment'] = False
+        ret['authenticated_user_pk'] = None
+
+        ret['has_viewing_permission'] = False
+        ret['has_rating_permission'] = False
+        ret['has_changing_permission'] = False
+        ret['has_deleting_permission'] = False
+        ret['has_moderating_permission'] = False
+        ret['has_comment_commenting_permission'] = False
+        if request and hasattr(request, 'user'):
+            user = request.user
+            ret['is_users_own_comment'] = (user.pk == instance.creator.pk)
+            ret['authenticated_user_pk'] = user.pk
+
+            ret['has_viewing_permission'] = user.has_perm(
+                'a4comments.view_comment', instance
+            )
+            ret['has_rating_permission'] = user.has_perm(
+                'a4comments.rate_comment', instance
+            )
+            ret['has_changing_permission'] = user.has_perm(
+                'a4comments.change_comment', instance
+            )
+            ret['has_deleting_permission'] = user.has_perm(
+                'a4comments.delete_comment', instance
+            )
+            ret['has_moderating_permission'] = user.has_perm(
+                'a4comments.moderate_comment', instance
+            )
+            ret['has_comment_commenting_permission'] = user.has_perm(
+                'a4comments.comment_comment', instance
+            )
+
```

Comment:
It feels like we talked about this before, but can you explain to me again why we add all of these here instead of using own fields?

Reply:
@Rineee Because we only need them in the representation, I thought it would be better this way. Instead of getting the user from the request in sooo many own fields, we do that only once, saving many lines of basically the same code.

Reply:
An idea could be to add them all in one field permissions?

Reply:
Yeah, I also thought about that. That we don't use the to_representation if we do not really really need to. 
But I still think this is better than adding all the fields to the top and then again at the bottom as get method.